app/Http/Controllers/MappingController.php

```diff
@@ -80,24 +99,42 @@ public function edit(Webhook $webhook, Mapping $mapping)
      * @param  Webhook $webhook, Mapping $mapping
      * @return \Illuminate\Http\Response
      */
-    public function update(MappingUpdateRequest $request, Webhook $webhook, Mapping $mapping)
+    public function update(MappingCreateRequest $request, $webhookId)
     {
-        $this->authorize('update', [$mapping, $webhook]);
-
-        $attributes = $request->except('_token');
-        $attributes['webhook_id'] = $webhook->id;
+        $webhook = Webhook::findOrFail($webhookId);
+        $this->authorize('update', $webhook);
 
+        DB::beginTransaction();
         try {
-            $mapping = $this->mappingRepository->update($mapping->id, $attributes);
+            $mappings = $webhook->mappings()->delete();
+            $keys = $request->keys;
+            $values = $request->values;
+            if ($keys && $values) {
+                for ($i = 0; $i < count($keys); $i++) {
+                    $attributes['webhook_id'] = $webhookId;
+                    $attributes['key'] = $keys[$i];
+                    $attributes['value'] = $values[$i];
+                    $mapping = Mapping::where('webhook_id', $webhookId)->updateOrCreate([
+                        'key' => $attributes['key']
```

Comment:
**Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Arrays.ArrayDeclaration.NoComma</br>**Severity**: error</br>**File**: app/Http/Controllers/MappingController.php</br>**Line**: 118</br>Each line in an array declaration must end in a comma